spyre/server.py

```diff
@@ -18,37 +18,50 @@
 templateLoader = jinja2.FileSystemLoader( searchpath=ROOT_DIR )
 templateEnv = jinja2.Environment( loader=templateLoader )
 
-
-
 class Root(object):
-	def __init__(self,templateVars=None, title=None, inputs=None, outputs=None, controls=None, tabs=None, getJsonDataFunction=None, getDataFunction=None, getTableFunction=None, getPlotFunction=None, getImageFunction=None, getD3Function=None, getCustomCSSFunction=None, getCustomJSFunction=None, getHTMLFunction=None,  getDownloadFunction=None, noOutputFunction=None):
+	def __init__(
+			self,
+			templateVars=None,
+			title=None,
+			inputs=None,
+			outputs=None,
+			controls=None,
+			tabs=None,
+			getJsonDataFunction=None,
+			getDataFunction=None,
+			getTableFunction=None,
+			getPlotFunction=None,
+			getImageFunction=None,
+			getD3Function=None,
+			getCustomCSSFunction=None,
+			getCustomJSFunction=None,
+			getHTMLFunction=None,
+			getDownloadFunction=None,
+			noOutputFunction=None,
+			):
 		# populate template dictionary for creating input,controler, and output HTML and javascript
 		if templateVars is not None:
 			self.templateVars = templateVars
 		else:
 			self.templateVars = {}
-			if title is not None:
-				self.templateVars['title'] = title
-			if inputs is not None:
-				self.templateVars['inputs'] = inputs
-			if controls is not None:
-				self.templateVars['controls'] = controls
-			if outputs is not None:
-				self.templateVars['outputs'] = outputs
-			if tabs is not None:
-				self.templateVars['tabs'] = tabs
-
-		self.getJsonData = getJsonDataFunction
-		self.getData = getDataFunction
-		self.getTable = getTableFunction
-		self.getPlot = getPlotFunction
-		self.getImage = getImageFunction
-		self.getD3 = getD3Function
-		self.getCustomJS = getCustomJSFunction
-		self.getCustomCSS = getCustomCSSFunction
-		self.getHTML = getHTMLFunction
-		self.noOutput = noOutputFunction
-		self.getDownload = getDownloadFunction
+			# Include any class arguments that should be added to the templateVars dictionary
+			templateVarsArgs = (
+					'title',
+					'inputs',
+					'controls',
+					'outputs',
+					'tabs'
+					)
+			for arg in templateVarsArgs:
+				if locals()[arg] is not None:
+					self.templateVars[arg] = locals()[arg]
+
+		# Adds any class arguments that have 'Function' to the object, with an attribute name omitting 'Function'
+		for key in locals().keys():
+			if 'Function' in key:
+				objAttributeName = key.replace('Function','')
+				setattr(self,objAttributeName,locals()[key])
+
 		d3 = self.getD3()
```

Comment:
i'm down with the change to templateVars. 

The self.getFoo = getFooFunction is really clunky, but it's more clear to anyone who's never seen this code before (and to me 6 months from now) what's going on. I'll add some comments too to make it even more clear.